### DISCUSSION TITLE
Minimum State Of Charge [%]

### DISCUSSION THREAD

**Discussion by xxxxminingxxxx** [wlcrs/huawei_solar#194] [Ideas]
First of all, thank you for the wonderful integration. 

Would there be a possibility for changing the value Minimum State Of Charge [%]?
When the days are shorter, I usually put it higher because the battery will not be fully charged, then the battery does not have to discharge so far. 
Now I have to log in to the inverter for this and it would of course be easier to do this from HA.

**Comment by Ptrancoso84** [comment]
hello
yes you can in "Battery End-of-discharge SOC" in the batery integration

![image](https://user-images.githubusercontent.com/117102688/212017549-04fb48e0-23ae-4a43-9fc4-598a8a752ec5.png)


be aware that, with the latest beta by default its configured to min 12% 

if you want less, you must change the values in the file settings in /config/custom_components/huawei_solar
services.yaml - >
forcible_charge_soc -> min: 12
forcible_discharge_soc: -> min: 12

![image](https://user-images.githubusercontent.com/117102688/212017461-714212f3-00f2-4eab-97b6-5051d2a019a0.png)

**Comment by JFHCFC** [comment]
> First of all, thank you for the wonderful integration.
> 
> Would there be a possibility for changing the value Minimum State Of Charge [%]? When the days are shorter, I usually put it higher because the battery will not be fully charged, then the battery does not have to discharge so far. Now I have to log in to the inverter for this and it would of course be easier to do this from HA.

Do you have off-grid options enabled? (in fusion solar app)
